### PR TITLE
[Preview] AI Integration: Adds Azure Core file in SDK to support diagnostics Scope

### PR DESCRIPTION
**Pull request by sourabh1007**
## Description

Copied Diagnostic Scope files from Azure Core to get the support of `DiagnosticScope `which will be helpful to create activity.

## Type of change
- [] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/OSS/Azure.Core/DiagnosticScopeFactory.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#nullable enable
+
+namespace Azure.Core.Pipeline
+{
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Threading;
+
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
+    internal class DiagnosticScopeFactory
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/OSS/Azure.Core/DiagnosticScopeFactory.cs:16]: Should there be tests validating these files?

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/OSS/Azure.Core/DiagnosticScopeFactory.cs:16]: Do we have tests for the other OSS embeds we have like Antlr or the Histogram?

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/OSS/Azure.Core/DiagnosticScopeFactory.cs (hunk 1 above)
Comment (FabianMeiswinkel) [reply Microsoft.Azure.Cosmos/src/OSS/Azure.Core/DiagnosticScopeFactory.cs:16]: No, we don't have any for those shaded dependencies (at least for Histogram) - and I think as long as we keep the files unmodified that is fine.
Integration tests would at least test the main scenarios - so I am good with this.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Review by j82w — APPROVED** [review]

**Pull request merged**